core_backend/app/llm_call/entailment.py

```diff
@@ -1,24 +1,38 @@
-from json import loads
-from typing import Any, Dict, Optional
+from typing import Any, Dict, List, Optional
+
+from pydantic import ValidationError
 
 from ..config import LITELLM_MODEL_URGENCY_DETECT
-from .llm_prompts import get_urgency_detection_prompt
+from ..utils import setup_logger
+from .llm_prompts import UrgencyDetectionEntailment
 from .utils import _ask_llm_async
 
+logger = setup_logger()
+
 
 async def detect_urgency(
-    urgency_rule: str, message: str, metadata: Optional[dict] = None
+    urgency_rules: List[str], message: str, metadata: Optional[dict] = None
 ) -> Dict[str, Any]:
     """
-    Detects the urgency of a message based on a given urgency rule.
+    Detects the urgency of a message based on a set of urgency rules.
     """
 
-    prompt = get_urgency_detection_prompt(urgency_rule, message)
-    json = await _ask_llm_async(
-        question="",
+    "\n".join([f"{i+1}. {rule}" for i, rule in enumerate(urgency_rules)])
+
+    ud_entailment = UrgencyDetectionEntailment(urgency_rules=urgency_rules)
+    prompt = ud_entailment.get_prompt()
+
+    json_str = await _ask_llm_async(
+        question=message,
         prompt=prompt,
         litellm_model=LITELLM_MODEL_URGENCY_DETECT,
         metadata=metadata,
     )
```

Comment:
You can also add `json=true` to optionally use OpenAI's json response format!

Reply:
Will it just get ignored if it's an LLM that doesn't use this?

Reply:
It should get ignored! passing json=true will pass an additional kwarg `response_format`
https://docs.litellm.ai/docs/completion/input